spec/rubocop/cop/lint/deprecated_class_methods_spec.rb

```diff
@@ -3,38 +3,64 @@
 RSpec.describe RuboCop::Cop::Lint::DeprecatedClassMethods do
   subject(:cop) { described_class.new }
 
-  it 'registers an offense for File.exists?' do
-    expect_offense(<<-RUBY.strip_indent)
-      File.exists?(o)
-           ^^^^^^^ `File.exists?` is deprecated in favor of `File.exist?`.
-    RUBY
-  end
+  before { inspect_source(source) }
+
+  shared_examples 'code with offense' do |code, highlights, expected|
```

Comment:
This looks like a re-implementation of `expect_offense`. Can’t we please use `expect_offense` instead?

Same comment applies to `'code without offense'` and `expect_no_offenses`.

Reply:
Thanks. I did not know about `expect_offense`. I simply searched for shared examples and found [these](https://github.com/rubocop-hq/rubocop/blob/7a1baaee07129123b0a4d6cc283064f6bda30838/spec/rubocop/cop/lint/empty_when_spec.rb#L10-L29) and used the same. I'll change them.

Reply:
Thanks. I’ll take a look at those other specs :-)

Reply:
Changed.